### PR TITLE
Fix Auth API behavior with tokens

### PR DESCRIPTION
**Pull request by kiootic**

### CODE REVIEW

File: pkg/auth/dependency/oauth/handler/result_authz.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type (
 func (a authorizationResultRedirect) WriteResponse(rw http.ResponseWriter, r *http.Request) {
 	query := a.RedirectURI.Query()
 	for k, v := range a.Response {
-		query.Set(k, v)
+		query.Add(k, v)
```
Comment (louischan-oursky) [inline pkg/auth/dependency/oauth/handler/result_authz.go]: Not sure why this change is related to the new test. Also this code snippet appears more than once. It is better to move this logic to `pkg/core/url/add_map.go`.

```golang
func AddMap(u *url.URL, m map[string]string) *url.URL {
  q := u.Query()
  for k, v := range m {
    q.Add(k, v)
  }
  out := *u
  out.RawQuery = q.Encode()
  return &out
}
```

Comment (kiootic) [reply pkg/auth/dependency/oauth/handler/result_authz.go]: The expected behavior is that the existing query parameters should be preserved, so `Add` should be used instead of `Set`. The test is there to prevent regression in case we forget it.
